### DISCUSSION TITLE
NextJS with Firebase Hosting vs Vercel

### DISCUSSION THREAD

**Discussion by IndigoBee** [firebase/firebase-tools#6665] [General]
I've just deployed the same app on Vercel and Firebase Hosting. Why is it much slower with Firebase Hosting? (mostly moving from one route to another)

Looking at it, it seems like with Firebase every time we change route it loads the new page, while with Vercel it already has loaded and cached everything on the first page load. How to fix this with Firebase Hosting?

We can see below that the content type of the files loaded for the other routes is defined as "text/html" for Firebase, and "text/x-component" for Vercel. 

Firebase Hosting:
![CleanShot 2024-01-06 at 17 12 19@2x](https://github.com/firebase/firebase-tools/assets/33088396/85bfde87-8fdd-476c-8187-4cbdd6056c1e)

Vercel:
![CleanShot 2024-01-06 at 17 11 19@2x](https://github.com/firebase/firebase-tools/assets/33088396/154f8e77-d3dd-4d2e-8d7c-f7c43e1a874c)

**Comment by crystalveey** [comment]
Can you explain better?
